system-design/class-diagram/domain/repository.puml

```diff
@@ -0,0 +1,42 @@
+@startuml
+package repository {
+    interface IBookRepository {
+        + GetBookByIsbn(long isbn): Task<Book?>
+        + GetBooksByTitle(string title, int offset, int limit): Task<IEnumerable<Book>>
+        + GetBooksByAuthor(string author, int offset, int limit): Task<IEnumerable<Book>>
+        + GetBooksByGenre(BookGenre genre, int offset, int limit): Task<IEnumerable<Book>>
+        + GetBooksByPublicationYear(DateTime publicationYear, int offset, int limit): Task<IEnumerable<Book>>
+    }
+    IBorrowRepository *-- BookOptions
+
+    interface IPatronRepository {
+        + GetPatronsByNameAsync(string name, int offset, int limit): Task<IEnumerable<Patron>>
+        + GetPatronByMembershipAsync(long membership): Task<Patron?>
+        + GetPatronsByContactDetailsAsync(long contactDetails, int offset, int limit): Task<IEnumerable<Patron>>
+    }
+    IPatronRepository *-- PatronOptions
```

Comment:
I think that the relation is Agregation instead of Composition, im not sure

Reply:
PatronOptions class has been deleted